src/oc/web/actions/activity.cljs

```diff
@@ -0,0 +1,125 @@
+            [oc.web.urls :as oc-urls]
+            [oc.web.router :as router]
+            [oc.web.dispatcher :as dis]
+            [oc.web.lib.utils :as utils]
+            [oc.web.actions :as actions]
+            [oc.web.lib.user-cache :as uc]
+            [oc.web.lib.responsive :as responsive]))
+
+(defn load-cached-item
+  [entry-data edit-key & [completed-cb]]
+  (let [cache-key (actions/get-entry-cache-key (:uuid entry-data))]
+    (uc/get-item cache-key
+     (fn [item err]
+       (if (and (not err)
+                (map? item)
+                (= (:updated-at entry-data) (:updated-at item)))
+         (let [entry-to-save (merge item (select-keys entry-data [:links :board-slug :board-name]))]
+           (dis/dispatch! [:input [edit-key] entry-to-save]))
+         (do
+           ;; If we got an item remove it since it won't be used
+           ;; since we have an updated version of it already
+           (when item
+             (actions/remove-cached-item (:uuid entry-data)))
+           (dis/dispatch! [:input [edit-key] entry-data])))
+       (when (fn? completed-cb)
+         (completed-cb))))))
+
+(defn activity-modal-fade-in
+  [activity-data & [editing item-load-cb]]
+  (let [org (router/current-org-slug)
+        board (:board-slug activity-data)
+        activity-uuid (:uuid activity-data)
+        to-url (oc-urls/entry board activity-uuid)]
+    (.pushState (.-history js/window) #js {} "" to-url)
+    (router/set-route! [org board activity-uuid "activity"]
+     {:org org
+      :board board
+      :activity activity-uuid
+      :query-params (dissoc (:query-params @router/path) :ap-initial-at)
+      :from-all-posts (= (router/current-board-slug) "all-posts")}))
+  (when editing
+    (utils/after 100 #(load-cached-item activity-data :modal-editing-data item-load-cb)))
+  (dis/dispatch! [:activity-modal-fade-in activity-data editing]))
+
+(defn activity-modal-fade-out
+  [board-slug board-filters]
+  (let [from-all-posts (:from-all-posts @router/path)
+        to-board (if from-all-posts "all-posts" board-slug)
+        org (router/current-org-slug)
+        to-url (if (string? board-filters)
+                (oc-urls/board-filter-by-topic org board-slug board-filters)
+                (if (= board-filters :by-topic)
+                  (oc-urls/board-sort-by-topic org board-slug)
+                  (if from-all-posts
+                    (oc-urls/all-posts org)
+                    (oc-urls/board org board-slug))))]
+    (.pushState (.-history js/window) #js {} "" to-url)
+    (router/set-route! [org to-board (if from-all-posts "all-posts" "dashboard")]
+     {:org org
+      :board to-board
+      :activity nil
+      :query-params (:query-params @router/path)
+      :from-all-posts false}))
+  (dis/dispatch! [:activity-modal-fade-out board-slug board-filters]))
+
+(defn entry-edit
+  [initial-entry-data]
+  (load-cached-item initial-entry-data :entry-editing)
+  (dis/dispatch! [:entry-edit initial-entry-data]))
```

Comment:
I can't see anything that handles this dispatch. I don't see anything that does in mainline either.

![screenshot](https://user-images.githubusercontent.com/55220/35749444-89ad27f4-081f-11e8-9d48-81782d6d56af.png)